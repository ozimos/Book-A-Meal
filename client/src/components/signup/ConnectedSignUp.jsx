import React from 'react';
import PropTypes from 'prop-types';
import Formsy from 'formsy-react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import isEqual from 'lodash.isequal';
import { userActions } from '../../redux/actions';
import MyInput from '../helpers/MyInput';
import MyCheckBox from '../helpers/MyCheckBox';
import '../../../public/styles/auth.scss';


class SignUp extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      canSubmit: false,
      newError: false,
      prevSignupError: {}
    };
    this.disableButton = this.disableButton.bind(this);
    this.enableButton = this.enableButton.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.serverFeedback = this.serverFeedback.bind(this);
    this.formEl = null;
  }
  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.signupError
      && !isEqual(nextProps.signupError, prevState.prevSignupError)) {
      return {
        prevSignupError: nextProps.signupError,
        newError: true
      };
    }
    return null;
  }
  componentDidUpdate() {
    const { prevSignupError, newError } = this.state;

    if (newError && Object.keys(prevSignupError).length !== 0) {
      this.serverFeedback(prevSignupError);
      // eslint-disable-next-line
      this.setState({ newError: false });
    }
  }
  handleSubmit(user) {
    this.setState({ prevSignupError: {} });
    this.props.dispatch(userActions.logout(user));
    this.props.dispatch(userActions.signUp(user));
  }
  disableButton() {
    this.setState({ canSubmit: false });
  }

  enableButton() {
    this.setState({ canSubmit: true });
  }
  serverFeedback(error) {
    this.formEl.updateInputsWithError(error);
  }


  render() {
    return (
      <div className="canvas">
        <div className="container2 container2-height">
          <header className="header">
            <h2 className="heading">Book A Meal</h2>
          </header>
          <main>
            <div className="welcome">
              <h4 className="welcome-text">
                Welcome!
              </h4>
              <h5>
                Register by entering the information below
              </h5>
            </div>
            <div className="form-box">
              <Formsy
                className="form"
                onValidSubmit={this.handleSubmit}
                onValid={this.enableButton}
                onInvalid={this.disableButton}
                ref={(form) => { this.formEl = form; }}
              >

                <MyInput
                  typeOfInput="email"
                  name="email"
                  required
                  placeholder="Email"
                  validations={{ isEmail: true, minLength: 5, maxLength: 48 }}
                  validationError="This is not a valid email"
                  validationErrors={{
                    minLength: 'input must be longer than 5 character',
                    maxLength: 'input must be shorter than 50 characters',
                    isEmail: 'This is not a valid email'
                  }}
                />
                <div className="invalid-feedback" />
                <MyInput
                  typeOfInput="text"
                  name="firstName"
                  placeholder="First Name"
                  validations={{ minLength: 1, maxLength: 48 }}
                  validationError="Please enter your first name"
                  validationErrors={{
                    minLength: 'input must be longer than 1 character',
                    maxLength: 'input must be shorter than 50 characters',
                  }}
                  required
                />
                <div className="invalid-feedback" />
                <MyInput
                  typeOfInput="text"
                  name="lastName"
                  placeholder="Last Name"
                  validations={{ minLength: 1, maxLength: 48 }}
                  validationError="Please enter your last name"
                  validationErrors={{
                    minLength: 'input must be longer than 1 character',
                    maxLength: 'input must be shorter than 50 characters',
                  }}
                  required
                />
                <MyInput
                  typeOfInput="password"
                  name="password"
                  placeholder="Password"
                  required
                  validations={{
                    hasUpperCase: (values, value) =>
                    /[A-Z]+|.{16,}/.test(value),
                    hasLowerCase: (values, value) =>
                    /[a-z]+|.{16,}/.test(value),
                    hasNumber: (values, value) =>
                    /\d+|.{16,}/.test(value),
                    hasSpecialCharacter: (values, value) =>
                    /\W+|.{16,}/.test(value),
                    minLength: 8,
                    maxLength: 48
                  }}
                  validationErrors={{
                    hasUpperCase: `Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters`,
                    hasLowerCase: `Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters`,
                    hasNumber: `Must have a number or you can
                     use a passphrase of minimum length 16 characters`,
                    hasSpecialCharacter: `Must have a special character or 
                    you can use a passphrase of minimum length 16 characters`,
                    minLength: 'Must have at least 8 characters',
                    maxLength: 'input must be shorter than 50 characters',
                  }}
                />
                <MyInput
                  typeOfInput="password"
                  name="confirmPassword"
                  placeholder="Confirm Password"
                  required
                  validations={{
                    equalsField: 'password',
                    minLength: 8,
                    maxLength: 48
                  }}
                  validationError="Passwords do not match"
                  validationErrors={{
                    equalsField: 'Passwords do not match',
                    minLength: 'Must have at least 8 characters',
                    maxLength: 'input must be shorter than 50 characters',
                  }}
                />
                <MyCheckBox
                  name="isCaterer"
                />
                <div>
                  <label htmlFor="isCaterer">
                    <MyCheckBox
                      name="isCaterer"
                      id="isCaterer"
                    />
                    <span>Caterer</span>
                  </label>
                </div>

                <button
                  className={this.state.canSubmit ? 'btn' : 'btn btn-disabled'}
                  type="submit"
                  disabled={!this.state.canSubmit}
                >
                  Continue
                </button>
              </Formsy>
              <div className="stacked-text">
                <Link
                  to="/login"
                >
                  <p>Already have an account? Click here to sign in</p>
                </Link>
              </div>
            </div>
          </main>

        </div>
      </div>
    );
  }
}

SignUp.propTypes = {
  dispatch: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  signupError: state.signupReducer.signupError
});
export { SignUp };
export default connect(mapStateToProps)(SignUp);
