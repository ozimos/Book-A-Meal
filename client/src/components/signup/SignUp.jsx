import React from 'react';
import PropTypes from 'prop-types';
import Formsy from 'formsy-react';
import { hot } from 'react-hot-loader';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userActions } from '../../redux/actions';
import MyInput from '../helpers/MyInput';
import MyInputCheckBox from '../helpers/MyInputCheckBox';
import '../../../public/styles/book_a_meal.css';
import '../../../public/styles/auth.scss';


class SignUp extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      canSubmit: false
    };
    this.disableButton = this.disableButton.bind(this);
    this.enableButton = this.enableButton.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.serverFeedback = this.serverFeedback.bind(this);
    this.formEl = null;
  }
  componentDidUpdate() {
    const { signupError } = this.props;
    if (Object.keys(signupError).length !== 0) {
      this.serverFeedback(signupError);
    }
  }
  handleSubmit(user) {
    this.props.dispatch(userActions.logout(user));
    this.props.dispatch(userActions.signUp(user));
  }
  disableButton() {
    this.setState({ canSubmit: false });
  }

  enableButton() {
    this.setState({ canSubmit: true });
  }
  serverFeedback(error) {
    this.formEl.updateInputsWithError(error);
  }


  render() {
    return (
      <div className="canvas">
        <div className="container2">
          <header className="header">
            <h1 className="heading">Book A Meal</h1>
          </header>
          <main>
            <div className="welcome">
              <h1 className="welcome-text">
                Welcome!
              </h1>
              <h4>
                Register by entering the information below
              </h4>
            </div>
            <div className="form-box">
              <Formsy
                className="form"
                onValidSubmit={this.handleSubmit}
                onValid={this.enableButton}
                onInvalid={this.disableButton}
                ref={(form) => { this.formEl = form; }}
              >

                <MyInput
                  typeOfInput="email"
                  name="email"
                  required
                  placeholder="Email"
                  validations="isEmail"
                  validationError="This is not a valid email"
                />
                <div className="invalid-feedback" />
                <MyInput
                  typeOfInput="text"
                  name="firstName"
                  placeholder="First Name"
                  validations="minLength:1"
                  validationError="Please enter your first name"
                  required
                />
                <div className="invalid-feedback" />
                <MyInput
                  typeOfInput="text"
                  name="lastName"
                  placeholder="Last Name"
                  validations="minLength:1"
                  validationError="Please enter your last name"
                  required
                />
                <MyInput
                  typeOfInput="password"
                  name="password"
                  placeholder="Password"
                  required
                  validations={{
                    hasUpperCase: (values, value) => /[A-Z]+|.{16,}/.test(value),
                    hasLowerCase: (values, value) => /[a-z]+|.{16,}/.test(value),
                    hasNumber: (values, value) => /\d+|.{16,}/.test(value),
                    hasSpecialCharacter: (values, value) => /\W+|.{16,}/.test(value),
                    minLength: 8
                  }}
                  validationErrors={{
                    hasUpperCase: 'Must have uppercase letter or you can use a passphrase of minimum length 16 characters',
                    hasLowerCase: 'Must have lowercase letter or you can use a passphrase of minimum length 16 characters',
                    hasNumber: 'Must have a number or you can use a passphrase of minimum length 16 characters',
                    hasSpecialCharacter: 'Must have a special character or you can use a passphrase of minimum length 16 characters',
                    minLength: 'Must have at least 8 characters'
                  }}
                />
                <MyInput
                  typeOfInput="password"
                  name="confirmPassword"
                  placeholder="Confirm Password"
                  required
                  validations={{
                    matchPassword: (values, value) => values.password === value
                  }}
                  validationErrors={{
                    matchPassword: 'Passwords do not match',
                  }}

                />
                <div>
                  <label htmlFor="isCaterer">
                    <MyInputCheckBox
                      name="isCaterer"
                      id="isCaterer"
                    />
                    Caterer
                  </label>
                </div>

                <button
                  className={this.state.canSubmit ? 'btn' : 'btn btn-disabled'}
                  type="submit"
                  disabled={!this.state.canSubmit}
                >
                  Continue
                </button>
              </Formsy>
              <div className="stacked-text">
                <Link to="/login"><p>Already have an account? Click here to sign in</p></Link>
              </div>
            </div>
          </main>

        </div>
      </div>
    );
  }
}
SignUp.defaultProps = {
  signupError: {}
};
SignUp.propTypes = {
  dispatch: PropTypes.func.isRequired,
  signupError: PropTypes.objectOf(PropTypes.string),
};
const mapStateToProps = state => ({
  signupError: state.signupReducer.signupError
});
export { SignUp };
export default connect(mapStateToProps)(hot(module)(SignUp));
