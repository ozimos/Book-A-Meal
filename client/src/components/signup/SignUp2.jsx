import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { userActions } from '../../redux/actions';
import FormConnected from '../helpers/Form';
import '../../../public/styles/book_a_meal.css';
import '../../../public/styles/auth.scss';


class SignUp2 extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: {
        firstName: '',
        lastName: '',
        email: '',
        password: '',
        confirmPassword: '',
        isCaterer: false
      },
      // submitted: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(event) {
    const { name, value } = event.target;
    const { user } = this.state;
    this.setState({
      user: {
        ...user,
        [name]: value
      }
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    // this.setState({ submitted: true });
    const { user } = this.state;
    const { dispatch } = this.props;
    if (user.firstName && user.lastName && user.email && user.password) {
      dispatch(userActions.signUp(user));
    }
  }
  render() {
    const { user } = this.state;
    return (
      <div className="canvas">
        <div className="container2">
          <header className="header">
            <h1 className="heading">Book A Meal</h1>
          </header>
          <main>
            <div className="welcome">
              <h1 className="welcome-text">
                Welcome!
              </h1>
              <h4>
                Register by entering the information below
              </h4>
            </div>
            <div className="form-box">
              <FormConnected
                className="form"
                onSubmit={this.handleSubmit}
                onChange={this.handleChange}
              >
                <input
                  type="email"
                  name="email"
                  required
                  placeholder="Email"
                  onChange={this.handleChange}
                  value={user.email}
                />
                <div className="invalid-feedback" />
                <input
                  type="text"
                  name="firstName"
                  required
                  value={user.firstName}
                  placeholder="First Name"
                  onChange={this.handleChange}
                />
                <div className="invalid-feedback" />
                <input
                  type="text"
                  name="lastName"
                  value={user.lastName}
                  required
                  placeholder="Last Name"
                  onChange={this.handleChange}
                />
                <div className="invalid-feedback" />
                <input
                  type="password"
                  name="password"
                  required
                  value={user.password}
                  placeholder="Password"
                  onChange={this.handleChange}
                  minLength={6}
                  pattern="(?=.*\d)(?=.*[a-z]).{6,}"
                />
                <div className="invalid-feedback" />
                <small className="directions">
                 Must be at least 6 characters long, contain letters and numbers
                </small>
                <input
                  type="password"
                  name="confirmPassword"
                  value={user.confirmPassword}
                  required
                  placeholder="Confirm Password"
                  onChange={this.handleChange}
                  minLength={6}
                  pattern="(?=.*\d)(?=.*[a-z]).{6,}"
                />
                <div className="invalid-feedback" />
                <div>
                  <label htmlFor="isCaterer">
                    <input
                      type="checkbox"
                      name="isCaterer"
                      id="isCaterer"
                      value={user.isCaterer}
                      onChange={this.handleChange}
                    />
                    Caterer
                  </label>
                </div>

                <button className="btn" type="submit">
                  Continue
                </button>
              </FormConnected>
              <div className="stacked-text">
                <Link to="/login"><p>Already have an account? Click here to sign in</p></Link>
              </div>
            </div>
          </main>

        </div>
      </div>
    );
  }
}
SignUp2.propTypes = {
  dispatch: PropTypes.func.isRequired,
};
export { SignUp2 };
export default connect(state => state)(hot(module)(SignUp2));
