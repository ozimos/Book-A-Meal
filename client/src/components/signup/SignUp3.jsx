import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { Link } from 'react-router-dom';
import { ValidatorForm } from 'react-form-validator-core';
import { connect } from 'react-redux';
import { userActions } from '../../redux/actions';
import TextValidator from '../helpers/TextValidator';
import '../../../public/styles/book_a_meal.css';

ValidatorForm.addValidationRule('isPasswordMatch', (value) => {
  if (value !== this.state.user.password) {
    return false;
  }
  return true;
});

class SignUp3 extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: {
        firstName: '',
        lastName: '',
        email: '',
        password: '',
        confirmPassword: '',
        isCaterer: false
      },
      // submitted: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleChange(event) {
    const { name, value } = event.target;
    const { user } = this.state;
    this.setState({
      user: {
        ...user,
        [name]: value
      }
    });
  }

  handleSubmit(event) {
    event.preventDefault();

    // this.setState({ submitted: true });
    const { user } = this.state;
    const { dispatch } = this.props;
    if (user.firstName && user.lastName && user.email && user.password) {
      dispatch(userActions.signUp(user));
    }
  }
  render() {
    const { user } = this.state;
    return (
      <div className="canvas">
        <div className="container2">
          <header className="header">
            <h1 className="heading">Book A Meal</h1>
          </header>
          <main>
            <div className="welcome">
              <h1 className="welcome-text">
                Welcome!
              </h1>
              <h4>
                Register by entering the information below
              </h4>
            </div>
            <div className="form-box">
              <ValidatorForm
                className="form"
                ref="form"
                onSubmit={this.handleSubmit}
                debounceTime={400}
              >

                <TextValidator
                  type="email"
                  name="email"
                  placeholder="Email"
                  onChange={this.handleChange}
                  value={user.email}
                  validators={['required', 'isEmail']}
                  errorMessages={['this field is required', 'email is not valid']}
                />
                <TextValidator
                  type="text"
                  name="firstName"
                  value={user.firstName}
                  placeholder="First Name"
                  validators={['required']}
                  errorMessages={['this field is required']}
                  onChange={this.handleChange}
                />
                <div className="invalid-feedback" />
                <TextValidator
                  type="text"
                  name="lastName"
                  value={user.lastName}
                  placeholder="Last Name"
                  validators={['required']}
                  errorMessages={['this field is required']}
                  onChange={this.handleChange}
                />
                <TextValidator
                  type="password"
                  name="password"
                  value={user.password}
                  placeholder="Password"
                  validators={['required', 'matchReqexp:^((?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^a-zA-Z0-9]).{8,}|.{16,})$', 'minStringLength:6']}
                  errorMessages={[' Must be at least 6 characters long, and contain uppercase and lowercase letters and numbers and special character or must be a passphrase 16 or more characters long']}
                  onChange={this.handleChange}
                />
                <TextValidator
                  type="password"
                  name="confirmPassword"
                  value={user.confirmPassword}
                  placeholder="Confirm Password"
                  required
                  onChange={this.handleChange}
                  validators={['required', 'isPasswordMatch']}
                  errorMessages={['Passwords do not match']}
                />
                <div className="invalid-feedback" />
                <div>
                  <label htmlFor="isCaterer">
                    <input
                      type="checkbox"
                      name="isCaterer"
                      id="isCaterer"
                      value={user.isCaterer}
                      onChange={this.handleChange}
                    />
                    Caterer
                  </label>
                </div>

                <button className="btn" type="submit">
                  Continue
                </button>
              </ValidatorForm>
              <div className="stacked-text">
                <Link to="/login"><p>Already have an account? Click here to sign in</p></Link>
              </div>
            </div>
          </main>

        </div>
      </div>
    );
  }
}
SignUp3.propTypes = {
  dispatch: PropTypes.func.isRequired,
};
export { SignUp3 };
export default connect(state => state)(hot(module)(SignUp3));
