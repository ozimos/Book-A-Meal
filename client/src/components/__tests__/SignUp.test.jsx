/* global React:false, shallow:false toJson:false */
import { SignUp } from '../signup/ConnectedSignUp';

const props = {
  dispatch: jest.fn(),
};
describe('SignUp Component', () => {
  const setup = () => shallow(<SignUp {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect((wrapper)
      .exists(<form className="form" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call `handleSubmit`', () => {
    const wrapper = setup();

    const handleSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'handleSubmit'
    );
    wrapper.instance().handleSubmit();
    expect(handleSubmitSpy).toHaveBeenCalled();
  });
  it('should call `disableButton`', () => {
    const wrapper = setup();

    const disableButtonSpy = jest.spyOn(
      wrapper.instance(),
      'disableButton'
    );
    wrapper.instance().disableButton();
    expect(disableButtonSpy).toHaveBeenCalled();
  });
  it('should call `enableButton`', () => {
    const wrapper = setup();

    const enableButtonSpy = jest.spyOn(
      wrapper.instance(),
      'enableButton'
    );
    wrapper.instance().enableButton();
    expect(enableButtonSpy).toHaveBeenCalled();
  });
});

