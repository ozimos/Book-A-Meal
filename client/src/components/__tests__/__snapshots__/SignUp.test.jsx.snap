// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders correctly 1`] = `
<div
  className="canvas"
>
  <div
    className="container2"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Register by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={true}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={Object {}}
            validations="isEmail"
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="firstName"
            placeholder="First Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your first name"
            validationErrors={Object {}}
            validations="minLength:1"
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="lastName"
            placeholder="Last Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your last name"
            validationErrors={Object {}}
            validations="minLength:1"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can use a passphrase of minimum length 16 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "minLength": 8,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="confirmPassword"
            placeholder="Confirm Password"
            required={true}
            typeOfInput="password"
            validationError="Passwords do not match"
            validationErrors={Object {}}
            validations="equalsField:password"
          />
          <div>
            <label
              htmlFor="isCaterer"
            >
              <Formsy(CheckBox)
                id="isCaterer"
                innerRef={[Function]}
                name="isCaterer"
                required={false}
                validationError=""
                validationErrors={Object {}}
                validations={null}
              />
              <span>
                Caterer
              </span>
            </label>
          </div>
          <button
            className="btn btn-disabled"
            disabled={true}
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/login"
          >
            <p>
              Already have an account? Click here to sign in
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;
